spec/services/towing_task_service_spec.rb

```diff
@@ -0,0 +1,45 @@
+describe TowingTaskService do
+  include ::Spec::Support::TenantIdentity
+
+  let(:source) { FactoryBot.create(:source, :tenant => tenant) }
+  let(:params) { {'source_id' => source.id} }
+  let(:subject) { described_class.new(params) }
+  let!(:launch_job_task) { LaunchJobTask.create!(:name => "task", :state => "running", :status => "ok", :tenant => tenant, :source => source, :input => input, :output => {"url" => "api/v1/url"}) }
+  let(:time_interval) { ClowderConfig.instance["INACTIVE_TASK_REMINDER_TIME"] * 60 }
+  let(:input) { {"jobs" => [{"method" => "launch", "href_slug" => "api/v2/job_templates/5/launch/"}]} }
+
+  around do |example|
+    Insights::API::Common::Request.with_request(default_request) { example.call }
+  end
+
+  describe "#process" do
+    let(:service) { instance_double(MQTTControllerService) }
+
+    before do
+      Timecop.safe_mode = true
+      allow(MQTTControllerService).to receive(:new).and_return(service)
+      allow(service).to receive(:process)
+    end
+
+    it "starts a new retry task" do
+      Timecop.travel(Time.current + time_interval) do
+        subject.process
+        launch_job_task.reload
+
+        expect(launch_job_task.message).to eq("Inactive, need towing task to reactivate")
+        expect(Task.find_by(:child_task_id => launch_job_task.id).input["jobs"].first["method"]).to eq("monitor")
```

Comment:
@hsong-rh Should we also check the URL in the Task is the url that we got back in output